File: DMS/app/src/main/java/com/example/dms/ChangePassword.java
package com.example.dms;

import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class ChangePassword extends AppCompatActivity {
    private MySqlite sqlite;
    private SQLiteDatabase myDb;
    private String id;
    private String type;
    private String password=null;
    private String newPassword=null;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.chenge_password);
        sqlite = new MySqlite(this, "Dormitory.db", null, 1);
        myDb = sqlite.getWritableDatabase();//打开数据库myDb

        Intent intent = getIntent();
        id=intent.getStringExtra("id");
        type = intent.getStringExtra("type");

        final EditText originalPassword = (EditText)findViewById(R.id.originalPassword);
        final EditText newPassword_1=(EditText)findViewById(R.id.newPassword1);
        final EditText newPassword_2=(EditText)findViewById(R.id.newPassword2);

        Cursor cursor;
        if(type.equals("admin")){
            cursor = myDb.rawQuery("select * from Admin where AdID=?",new String[]{id});
            if(cursor.moveToFirst()){
                password  = cursor.getString(Integer.parseInt(String.valueOf(cursor.getColumnIndex("PassWord"))));
            }
        }
        else if(type.equals("student")){
            cursor = myDb.rawQuery("select * from User where UserID=?",new String[]{id});
            if(cursor.moveToFirst()){
                password  = cursor.getString(Integer.parseInt(String.valueOf(cursor.getColumnIndex("Password"))));
            }
        }
        else{
            cursor = myDb.rawQuery("select * from General_Admin where GAID=?",new String[]{id});
            if(cursor.moveToFirst()){
                password  = cursor.getString(Integer.parseInt(String.valueOf(cursor.getColumnIndex("password"))));
            }
        }

        Log.d("SSS","dwad"+password);

        Button ok = (Button)findViewById(R.id.ok);
        ok.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(originalPassword.getText().toString().equals(password) && newPassword_1.getText().toString().equals(newPassword_2.getText().toString())){
                    newPassword=newPassword_1.getText().toString();
                    String sqliteUpdate = null;
                    if(type.equals("admin")){
                        sqliteUpdate="UPDATE Admin SET PassWord ="+newPassword+" WHERE AdID="+id;
                        myDb.execSQL(sqliteUpdate);
                    }else if(type.equals("student")){
                        sqliteUpdate="UPDATE User SET Password ="+newPassword+" WHERE UserID="+id;
                        myDb.execSQL(sqliteUpdate);
                    }
                    else{
                        sqliteUpdate="UPDATE General_Admin SET password ="+newPassword+" WHERE GAID="+id;
                        myDb.execSQL(sqliteUpdate);
                    }
                    Toast toast= Toast.makeText(ChangePassword.this,"更改成功！",Toast.LENGTH_SHORT);
                    toast.show();
                    finish();
                }
                else{
                    if(!originalPassword.equals(password))
                        Toast.makeText(ChangePassword.this,"原密码错误！",Toast.LENGTH_SHORT).show();
                    else
                        Toast.makeText(ChangePassword.this,"两次密码不相同！",Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: news/app/src/main/java/com/hjzf/ui/ChangePassword.kt
package com.app.ui.activity

import android.graphics.Color
import android.os.Bundle
import android.text.InputType
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.hjzf.R

class ChangePassword : AppCompatActivity(){

    override fun onCreate(savedInstanceState: Bundle?) {


        super.onCreate(savedInstanceState)

        setContentView(R.layout.change_password)
        val originpassword = findViewById<TextView>(R.id.originalPassword)
        val newpassword_1 = findViewById<TextView>(R.id.newPassword1)
        val newpassword_2 = findViewById<TextView>(R.id.newPassword2)
        val button = findViewById<Button>(R.id.ok)
        var  imageview1 = findViewById<ImageView>(R.id.hidePassword0)
        var  imageview2 = findViewById<ImageView>(R.id.hidePassword1)
        var  imageview3 = findViewById<ImageView>(R.id.hidePassword2)
        var  tv1 = findViewById<TextView>(R.id.originalPassword)
        var  tv2 = findViewById<TextView>(R.id.newPassword1)
        var  tv3 = findViewById<TextView>(R.id.newPassword2)


        val sqlite = Sqlite(this,"news_db",1)
        val id = intent.getStringExtra("id")
        var newpassword = "test"
        var password = "test"
        var i = 1
        var j = 1
        var k = 1
        var mydb = sqlite.writableDatabase
/*
        val cursor  =  mydb.rawQuery("select * from User where UserID =?", arrayOf(id))
        if(cursor.moveToFirst()){
            password = cursor.getString(cursor.getColumnIndex("Password"))
        }

 */
        button.setOnClickListener {
                 if (originpassword.text.toString().equals(password)&&newpassword_1.text.toString().equals(newpassword_2.text.toString())){
                     newpassword=newpassword_1.text.toString()
                    var sqliteUpdate ="UPDATE User SET Password =" +newpassword+"WHERE UserID=" +id
                    mydb.execSQL(sqliteUpdate)
                     Toast.makeText(this@ChangePassword, "更改成功！", Toast.LENGTH_SHORT).show()
                     finish()

                 }
                 else{
                    if (!originpassword.equals(password))
                        Toast.makeText(this@ChangePassword, "原密码错误！", Toast.LENGTH_SHORT).show()
                    else
                        Toast.makeText(this@ChangePassword, "两次密码不相同！", Toast.LENGTH_SHORT).show()
                 }
        }


        //隐藏密码，可选隐藏密码或者显示密码，两个状态，避免密码泄露
        imageview1.setOnClickListener {
            if (i%2!=0){
                imageview1.setColorFilter(Color.BLUE)
                tv1.setInputType(InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD)
            }
            else{
                imageview1.setColorFilter(Color.GRAY)
                tv1.setInputType(InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD)
            }
            i++

        }
        imageview2.setOnClickListener {
            if (j%2!=0){
                imageview2.setColorFilter(Color.BLUE)
                tv2.setInputType(InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD)
            }
            else{
                imageview2.setColorFilter(Color.GRAY)
                tv2.setInputType(InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD)
            }
            j++

        }
        imageview3.setOnClickListener {
            if (k%2!=0){
                imageview3.setColorFilter(Color.BLUE)
                tv3.setInputType(InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD)
            }
            else{
                imageview3.setColorFilter(Color.GRAY)
                tv3.setInputType(InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD)
            }
            k++
        }






    }


}